packages/core/src/storages/request_manager_tandem.ts

```diff
@@ -0,0 +1,162 @@
+import type { Dictionary } from '@crawlee/types';
+
+import type { Log } from '@apify/log';
+
+import { log } from '../log';
+import type { Request, Source } from '../request';
+import type { IRequestList } from './request_list';
+import type {
+    AddRequestsBatchedOptions,
+    AddRequestsBatchedResult,
+    IRequestManager,
+    RequestQueueOperationInfo,
+    RequestQueueOperationOptions,
+    RequestsLike,
+} from './request_provider';
+
+/**
+ * A request manager that combines a RequestList and a RequestQueue.
+ * It first reads requests from the RequestList and then, when needed,
+ * transfers them in batches to the RequestQueue.
+ */
```

Comment:
The description could be better. 

>  A request manager that ...

I believe "request manager" is a new term in this PR, right? Let's maybe stick to request provider, which (albeit not that popular), is already used in Crawlee? Also connected to:

> It first reads requests from the RequestList and then, when needed, transfers them in batches to the RequestQueue.

This sounds like transferring requests from one storage to another is the main use case for this class (it's providing requests to the crawler instance).